crates/signer/src/signer.rs

```diff
@@ -100,6 +101,17 @@ pub trait Signer: Send + Sync {
         self.sign_hash(payload.eip712_signing_hash(domain)).await
     }
 
+    /// Encodes and signs the typed data according to [EIP-712] for Signers that are not dynamically sized.
+    #[cfg(feature = "eip712")]
+    #[inline]
+    async fn sign_dynamic_typed_data(
+        &self,
+        payload: &TypedData
+    ) -> Result<Signature> {
+        let hash = payload.eip712_signing_hash().unwrap();
```

Comment:
Error should be propagated, not unwrapped. You need to add a variant to the `Error` enum https://github.com/alloy-rs/alloy/blob/7b35f83099f78dcebfb5d1075f2245a85380a774/crates/signer/src/error.rs#L11

Reply:
Resolved, thank you